js/lib/ilib.js

```diff
@@ -284,6 +284,43 @@ ilib.setLocale = function (spec) {
     // to look into them to find a locale
 };
 
+/**
+ * @private
+ */
+function parseLocale(str) {
+    if (!str || str === "C") return str;
+
+    // take care of the libc style locale with a dot + script at the end
+    var dot = str.indexOf('.')
+    if (dot > -1) {
+        str = str.substring(0, dot);
+    }
+
+    // handle the posix default locale
+    if (str === "C") return "en-US";
+
+    if (str.length >= 10) {
+        return [
+            str.substring(0,2).toLowerCase(),
+            str[3].toUpperCase() + str.substring(4,7).toLowerCase(),
+            str.substring(8,10).toUpperCase()
+        ].join("-");
+    }
+
+    if (str.length >= 5) {
+        return [
+            str.substring(0,2).toLowerCase(),
+            str.substring(3,5).toUpperCase()
+        ].join("-");
+    }
+
```

Comment:
So we assume any of the platforms returns locale with language-script form.  never happen to get something like zh-Hans.  we get language-country or language-script-country form all the time. am I right?

Reply:
Addressed each comment. Please re-review!